spec/spec.md

```diff
@@ -1390,28 +1390,60 @@ format-related rules above:
     },
     "field": {
       "type": "object",
-      "properties": {
-        "path": {
-          "type": "array",
-          "items": { "type": "string" }
+      "oneOf": [
+        {
+          "properties": {
+            "path": {
+              "type": "array",
+              "items": { "type": "string" }
+            },
+            "purpose": { "type": "string" },
+            "filter": {
+              "type": "object",
+              "properties": {
+                "type": { "type": "string" },
+                "format": { "type": "string" },
+                "pattern": { "type": "string" },
+                "minimum": { "type": "string" },
+                "minLength": { "type": "integer" },
+                "maxLength": { "type": "integer" }
+              },
+              "required": ["type"],
+              "additionalProperties": false
+            }
+          },
+          "required": ["path"],
+          "additionalProperties": false
         },
-        "purpose": { "type": "string" },
-        "filter": {
-          "type": "object",
+        {
           "properties": {
-            "type": { "type": "string" },
-            "format": { "type": "string" },
-            "pattern": { "type": "string" },
-            "minimum": { "type": "string" },
-            "minLength": { "type": "integer" },
-            "maxLength": { "type": "integer" }
+            "path": {
+              "type": "array",
+              "items": { "type": "string" }
+            },
+            "purpose": { "type": "string" },
+            "filter": {
+              "type": "object",
+              "properties": {
+                "type": { "type": "string" },
+                "format": { "type": "string" },
+                "pattern": { "type": "string" },
```

Comment:
I there is a `predicate` property, I'm not sure the `filter` property should allow `format` and `pattern`.
```suggestion
```

Reply:
I agree but that may make more sense to leave to an implementer.

Reply:
At first, I wondered if `format` and `pattern` should be excluded because they aren't used in any of the predicate operations. Now I wonder if they should be explicitly added in the examples, because asking for a the result of a predicate operation on a regex might be a really powerful capability (though I'm not sure how it might be done mathematically, cryptographically, and in zero-knowledge).
So, I'm also in favor of leaving them in at this moment, if only for the possibility of future coolness.